grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -660,9 +660,9 @@ class OrderController {
                 throw new UnsupportedOperationException("${warehouse.message(code: 'errors.noPermissions.label')}")
             }
             orderItem.properties = params
-            Shipment pendingShipment = order.pendingShipment
-            if (pendingShipment) {
-                Set<ShipmentItem> itemsToUpdate = pendingShipment.shipmentItems.findAll { it.orderItemId == orderItem.id }
+            List pendingShipments = order.pendingShipments
+            if (pendingShipments) {
+                def itemsToUpdate = pendingShipments*.shipmentItems*.findAll { it.orderItemId == orderItem.id }.flatten().toArray()
```

Comment:
I know I'm guilty of doing this too, but let's try to keep this kind gangly code in a method in the service or on the domain class for readability and so that it's easier to unit test. 

So add a pendingShipmentItems method to OrderItem
```
def pendingShipmentItems() { 
    return pendingShipments*.shipmentItems*.findAll { 
        it.orderItemId == orderItem.id 
    }.flatten().toArray()
}
```
As well as this one 
```
void refreshPendingShipmentItemRecipients() { 
    pendingShipmentItems.each { 
        it.recipient = orderItem.recipient
    }
}
```